alpha/apps/kaltura/lib/clipconcat/kClipManager.php

```diff
@@ -0,0 +1,509 @@
+		if ($internalResource instanceof kFileSyncResource && $internalResource->getOriginEntryId()) {
+			$this->createParentBatchJob($internalResource->getOriginEntryId(), $clipEntry, $dbEntry,
+													$dbEntry->getPartnerId(), $operationAttributes);
+		} else {
+			$this->createParentBatchJob(null, $clipEntry, $dbEntry, $dbEntry->getPartnerId(), $operationAttributes);
+		}
+	}
+
+	/***
+	 * @param $sourceFlavorParamId
+	 * @return int
+	 * @throws PropelException
+	 */
+	private function cloneFlavorParam($sourceFlavorParamId)
+	{
+		$flavorParamsObj = assetParamsPeer::retrieveByPK($sourceFlavorParamId);
+		// unset flavorParamsObj ID
+		$flavorParamsObj->setId(null);
+		$flavorParamsObj->setNew(true);
+		$flavorParamsObj->setFormat(flavorParams::CONTAINER_FORMAT_MPEGTS);
+		//save the object
+		$flavorParamsObj->save();
+		//return the object ID
+		return $flavorParamsObj->getId();
+	}
+
+
+	/**
+	 * @param string $entryId
+	 * @return flavorAsset
+	 */
+	private function setDummyOriginalFlavorAssetReady($entryId)
+	{
+
+		$flavorAsset = assetPeer::retrieveOriginalByEntryId($entryId);
+		//set Dummy Ready we will update it later
+		$flavorAsset->setStatus(flavorAsset::ASSET_STATUS_READY);
+		$flavorAsset->save();
+		return $flavorAsset;
+	}
+
+
+	/**
+	 * @param int $partnerId
+	 * @param string $entryId
+	 * @param $flavorParamId
+	 * @param int $order
+	 * @return flavorAsset
+	 */
+	private function createTempClipFlavorAsset($partnerId, $entryId, $flavorParamId,$order)
+	{
+		$entry = entryPeer::retrieveByPK($entryId);
+		if(!$entry)
+		{
+			KalturaLog::err("Entry [$entryId] not found");
+			return null;
+		}
+
+		// creates the flavor asset
+		$flavorAsset = flavorAsset::getInstance();
+		$flavorAsset->setStatus(flavorAsset::ASSET_STATUS_QUEUED);
+		$flavorAsset->incrementVersion();
+		$flavorAsset->addTags(array(flavorParams::TAG_TEMP_CLIP));
+		$flavorAsset->setIsOriginal(false);
+		$flavorAsset->setFlavorParamsId($flavorParamId);
+		$flavorAsset->setPartnerId($partnerId);
+		$flavorAsset->setEntryId($entryId);
+		$flavorAsset->save();
+		$flavorAsset->putInCustomData(self::CLIP_NUMBER,$order);
+
+		return $flavorAsset;
+	}
+
+
+
+
+	/**
+	 * @param BatchJob $batchJob
+	 * @throws Exception
+	 */
+	private function startConcat($batchJob)
+	{
+		KalturaLog::info('Going To Start Concat Job');
+		if($batchJob->getJobType() != BatchJobType::CLIP_CONCAT)
+		{
+			return;
+		}
+
+		foreach ($batchJob->getChildJobs() as $job)
+		{
+			/** @var BatchJob $job */
+			KalturaLog::info('Child job id [' . $job->getId() . '] status [' . $job->getStatus() . ']' . '] type ['.$job->getJobType() .']' );
+			KalturaLog::info('Flavor Param Ids:' .$job->getEntry()->getFlavorParamsIds());
+		}
+		/** @var kClipConcatJobData $jobData */
+		$jobData = $batchJob->getData();
+
+		$tempEntry = entryPeer::retrieveByPK($jobData->getTempEntryId());
+		$assets = assetPeer::retrieveByEntryId($jobData->getTempEntryId());
```

Comment:
Need to limit asset to the only **type** you know to concat which is flavorAsset

Reply:
$assets = assetPeer::retrieveByEntryId($jobData->getTempEntryId(), array(assetType::FLAVOR));